message_ix/model/MESSAGE-MACRO_run.gms

```diff
@@ -199,6 +202,10 @@ DISPLAY enestart, eneprice, total_cost ;
 * solve MACRO model                                                                                                    *
 *----------------------------------------------------------------------------------------------------------------------*
 
+* update total energy system costs by node and time with information from latest MESSAGE run
+total_cost(node_macro, year) = COST_NODAL_NET.L(node_macro, year) / 1000 ;
```

Comment:
I see that here we introduce a new MESSAGE parameter (`total_cost`) to report an existing variable (`COST_NODAL_NET`) only with a unit conversion. As the units are not explicit in MESSAGE, it is not clear to me the use case of this new parameter? In general, I believe it's better not to introduce too many parameters in the output file if the same information can be seen in another parameter/variable.

Reply:
`total_cost ` parameter is not introduced in this code but it is already introduced in `macro_data_load.gms` and used in macro_core.gms as follows:

```
COST_ENERGY(node_active, year) $ (NOT macro_base_period(year))..
EC(node_active, year) =E=
(total_cost(node_active, year)/1000
+ SUM(sector, eneprice(node_active, sector, year) * 1E-3 * (PHYSENE(node_active, sector, year) - enestart(node_active, sector, year)))
+ SUM(sector, eneprice(node_active, sector, year) * 1E-3 / enestart(node_active, sector, year) * (PHYSENE(node_active, sector, year) - enestart(node_active, sector, year)) * (PHYSENE(node_active, sector, year) - enestart(node_active, sector, year))))
;
```
Here the purpose of adding that line was not unit conversion but to update this parameter based on the COST_NODAL_NET in each iteration with MESSAGE.

Reply:
Thanks for the explanation.